### ISSUE TITLE
Overriding _ipython_display_ for widgets in a Box

### ISSUE THREAD

**Issue by andymcarter** [jupyter-widgets/ipywidgets#2274]
I have been subclassing the basic ipywidgets to develop notebook GUIs, and have been trying to 'wrap' a basic widget with a HTML label to the left (for consistent layouts). I'm aware there is a `description` kwarg but I often want to create a single class which contains multiple widgets (e.g. a start and end date picker), but just has a single label.

I understand `_ipython_display_` can be overrided to change how a widget is displayed (following the example [here](https://github.com/jupyter-widgets/ipywidgets/blob/1f93ad39802d3ee4a6b88bc13064f93f8f796374/docs/source/examples/Variable%20Inspector.ipynb)).

This works fine when calling a widget to display directly (see below):

```python
import ipywidgets as ipyw

label_layout = ipyw.Layout(width='100px')

class WrappedToggles(ipyw.ToggleButtons):
    
    def __init__(self):
        layout = ipyw.Layout(width='400px')
        super().__init__(options=['1','2'], layout=layout)

    def _ipython_display_(self, **kwargs):
        label = 'Label:'
        box_label = ipyw.HTML(value=label, layout=label_layout)
        children = (box_label, self)
        ipyw.HBox(children=children)._ipython_display_()

WrappedToggles()
```

![image](https://user-images.githubusercontent.com/22293319/49139516-eb131c00-f2e9-11e8-8800-9b8855648a29.png)

However, if I try to add these widgets to a Box, the `_ipython_display_` method is not respected, and the widget displays as if it is the simple inherited widget (in this case, the Togglebuttons).

``` python
children = [WrappedToggles(), WrappedToggles()]
ipyw.VBox(children=children)
```

![image](https://user-images.githubusercontent.com/22293319/49139221-1fd2a380-f2e9-11e8-814e-7e4fc97bff0c.png)

**Can I get the custom `_ipython_display_` behaviour when combining these 'wrapped' widgets into a dashboard Box?**

ipywidgets version 7.4.2
IPython 7.1.1

**Comment by maartenbreddels** [comment]
I don't think this should be supported, those are two different worlds, the display mechanism (out of the ipywidget realm) and the widgets. I think a better solution would be to inherit from HBox, and add the children. Would that work for you?

**Comment by SylvainCorlay** [comment]
The base `_ipython_display_` for widgets will be removed in the next major version and we will fully rely on `_repr_mimebundle_`.

**Comment by andymcarter** [comment]
Understood Maarten, thanks. Hadn't appreciated that `_ipython_display_` wasn't a specific thing for widgets.

I guess I should inherit from HBox, create HTML and ToggleButtons widgets in the `__init__`, then pass them as children to `super().__init__`.  Then I could override the required methods on this class to delegate to the ToggleButtons (like `get_interact_value()`).  Sounds like a more compositional approach would be beneficial here?

**Comment by vidartf** [comment]
Xref #1811.
